pkg/server/httpserver.go

```diff
@@ -84,6 +86,17 @@ func (s *HTTPServer) getProblemMessage(err error, context ...interface{}) string
 }
 
 // Run starts listening for incoming connetions
-func (s HTTPServer) Run() error {
+func (s *HTTPServer) Run(ctx context.Context) error {
+	// this function listens for finished context and calls graceful shutdown on the API server
+	go func() {
+		select {
+		case <-ctx.Done():
+			// sleep 2 seconds to cover the edge case if SIGTERM or SIGKILL signal occurs before the server is started,
+			// so the application does not get stuck
+			time.Sleep(2 * time.Second)
+			s.Log.Infof("shutting down Testkube API server")
+			_ = s.Mux.Shutdown()
```

Comment:
Doesit return error, should we ignore it?

Reply:
It is safe to ignore it as we cannot really do anything about it, the app is shutting down for some reason (sigterm, sigkill...)